src/lib/events.tsx

```diff
@@ -0,0 +1,97 @@
+import { ENV } from '../lib/env'
+
+const LOCAL_USER_ID_FUNCTION_HANDLER = 'local_fe_events_direct_user_id_handler'
+const LOCAL_FE_EVENTS_MOODLE_FUNCTION_HANDLER = 'local_fe_events_moodle_content_loaded_handler'
+const LOCAL_FE_EVENTS_DIRECT_FUNCTION_HANDLER = 'local_fe_events_direct_content_loaded_handler'
+
+const PIPELINE_MOODLE_EVENTS = 'moodle_eventing'
+const PIPELINE_MOODLE_DIRECT = 'moodle_direct'
+const PIPELINE_FE_DIRECT = 'fe_direct'
+
+function getMoodlePluginMethodUrl(endpointMethod: string): string {
+  if (window.M === undefined) {
+    console.log('moodle variables not found')
+    return ''
+  }
+  let userUrl: string
+  userUrl = String(window.M.cfg.wwwroot)
+  userUrl += '/lib/ajax/service.php?sesskey='
+  userUrl += String(window.M.cfg.sesskey)
+  userUrl += '&info=' + endpointMethod
+  return userUrl
+}
+
+async function sendEventMoodlePlugin(pipelineId: string): Promise<void> {
+  const contentId = 'moodle_test_content_via_moodle_direct'
+  let endpoint, url
+  if (pipelineId === PIPELINE_MOODLE_DIRECT) {
+    endpoint = LOCAL_FE_EVENTS_DIRECT_FUNCTION_HANDLER
+    url = getMoodlePluginMethodUrl(endpoint)
+  } else {
+    endpoint = LOCAL_FE_EVENTS_MOODLE_FUNCTION_HANDLER
+    url = getMoodlePluginMethodUrl(LOCAL_FE_EVENTS_MOODLE_FUNCTION_HANDLER)
+  }
+
+  const request = {
+    index: 0,
+    methodname: endpoint,
+    args: {
+      contentId: contentId
+    }
+  }
+
+  const response = await fetch(url, { method: 'POST', body: JSON.stringify([request]) })
+  if (response.ok) {
+    const responseJson = await response.json()
+    console.log(responseJson)
+    if (responseJson.error === undefined) {
+      console.log('response success')
+    } else {
+      console.log('response ERROR: ' + String(responseJson))
+    }
```

Comment:
I think this error handling is a bit broken. The AJAX endpoint seems to return an array of results, so something like the following might work better:

```javascript
    if (responseJson[0].error === true) {
      console.error(`Error from Moodle: ${JSON.stringify(responseJson[0])}`)
    }
```